### PR TITLE
Use Click for CLI, clean up of conf.py, remove unused workers

### PR DESCRIPTION
**Pull request by NicolasAubry**

### CODE REVIEW

File: tachyon/__main__.py [hunk 1]
```diff
@@ -145,31 +141,33 @@ def finish_output(print_worker):
         pass
 
 
-def main():
-    # Parse command line
-    from tachyon.core.arguments import generate_options, parse_args
```
Comment (lphuberdeau) [inline tachyon/__main__.py:150]: Is the file still in use? Shouldn't be removed as well?

File: tachyon/__main__.py [hunk 2]
```diff
@@ -145,31 +141,33 @@ def finish_output(print_worker):
...
-    if len(sys.argv) <= 1:
-        parser.print_help()
-        print('')
-        return
+@click.command()
+@click.option("--cookie-file", default="")
+@click.option("--depth-limit", default=2)
+@click.option("--directories-only", is_flag=True)
+@click.option("--files-only", is_flag=True)
+@click.option("--json-output", is_flag=True)
```
Comment (lphuberdeau) [inline tachyon/__main__.py]: Missing short versions?

File: tachyon/__main__.py [hunk 3]
```diff
@@ -145,31 +141,33 @@ def finish_output(print_worker):
...
+@click.option("--max-retry-count", default=3)
+@click.option("--plugins-only", is_flag=True)
+@click.option("--proxy", default="")
+@click.option("--recursive", is_flag=True)
+@click.option("--user-agent", default=default_user_agent)
+@click.option("--vhost", type=str, default=None)
+@click.argument("target_host")
+def main(target_host, cookie_file, json_output, max_retry_count,
```
Comment (lphuberdeau) [inline tachyon/__main__.py]: Missing -x (--plugin-configure)

File: tachyon/core/__version__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+# version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+# Place, Suite 330, Boston, MA  02111-1307  USA
+
+__version__ = "3.0"
```
Comment (lphuberdeau) [inline tachyon/core/__version__.py]: Should probably be 3.0.0

File: tachyon/__main__.py [hunk 5]
```diff
@@ -113,20 +109,20 @@ def load_execute_file_plugins():
...
-    if conf.cookies is not None:
-        set_cookies(hammertime, conf.cookies)
+async def scan(hammertime, *, cookies=None, directories_only=False, files_only=False, plugins_only=False, **kwargs):
+    if cookies is not None:
+        set_cookies(hammertime, cookies)
     else:
         await get_session_cookies(hammertime)
 
-    load_execute_host_plugins()
+    #load_execute_host_plugins()
```
Comment (lphuberdeau) [inline tachyon/__main__.py]: This seems like a mistake.

**Review by lphuberdeau — CHANGES_REQUESTED** [review]

File: tachyon/__main__.py (hunk 5 above)
Comment (NicolasAubry) [reply tachyon/__main__.py]: plugins rely on configuration in conf.py and no longer work now. Plugin migration will be done in the future.

File: tachyon/__main__.py (hunk 2 above)
Comment (NicolasAubry) [reply tachyon/__main__.py]: I don't like non-descriptive options such as -f, -d, or -l, but I can add them if it is required.

File: tachyon/__main__.py (hunk 1 above)
Comment (NicolasAubry) [reply tachyon/__main__.py:150]: I will

File: tachyon/__main__.py [hunk 6]
```diff
@@ -145,31 +141,33 @@ def finish_output(print_worker):
...
+@click.option("-j", "--json-output", is_flag=True)
+@click.option("-m", "--max-retry-count", default=3)
+@click.option("-z", "--plugins-only", is_flag=True)
+@click.option("-x", "--plugin-settings", default=[])
+@click.option("-p", "--proxy", default="")
+@click.option("-r", "--recursive", is_flag=True)
+@click.option("-u", "--user-agent", default=default_user_agent)
+@click.option("-v", "--vhost", type=str, default=None)
+@click.argument("target_host")
+def main(target_host, cookie_file, json_output, max_retry_count, plugin_settings, proxy, user_agent, vhost, **kwargs):
```
Comment (lphuberdeau) [inline tachyon/__main__.py]: Should begin with * to make sure it is not order-dependant.

All parameters should be handled.

File: tachyon/__main__.py [hunk 7]
```diff
@@ -145,31 +141,33 @@ def finish_output(print_worker):
...
-        return
+@click.command()
+@click.option("-c", "--cookie-file", default="")
+@click.option("-l", "--depth-limit", default=2)
+@click.option("-s", "--directories-only", is_flag=True)
+@click.option("-f", "--files-only", is_flag=True)
+@click.option("-j", "--json-output", is_flag=True)
+@click.option("-m", "--max-retry-count", default=3)
+@click.option("-z", "--plugins-only", is_flag=True)
+@click.option("-x", "--plugin-settings", default=[])
```
Comment (lphuberdeau) [inline tachyon/__main__.py]: http://click.pocoo.org/5/options/#multiple-options

File: tachyon/core/conf.py [hunk 8]
```diff
@@ -21,28 +21,12 @@
 # Internal config and consts
 name = "delvelabs/tachyon"
 version = '2.1.3'
```
Comment (lphuberdeau) [inline tachyon/core/conf.py]: Should be taken from __version__.

**Review by lphuberdeau — CHANGES_REQUESTED** [review]

**Review by lphuberdeau — APPROVED** [review]

**Pull request merged**